### PR TITLE
GMP Documentation, Alerts, and Dashboard filters: Kibana

### PR DESCRIPTION
**Pull request by algchoo**
This PR adds the following for the Kibana GMP integration:

* dashboard filters in `kibana-prometheus.json`
* adds alerts in `documentation.yaml` and `prometheus_metadata.yaml`

### CODE REVIEW

**Review by mtabasko — CHANGES_REQUESTED** [review]

File: integrations/kibana/documentation.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+app_site_url: https://www.elastic.co/kibana/
+exporter_name: the Kibana exporter
+exporter_pkg_name: kibana-prometheus-exporter
+exporter_repo_url: https://github.com/pjhampton/kibana-prometheus-exporter
+additional_prereq_info: |
+  {{app_name_short}} exposes Prometheus-format metrics when the [prometheus plugin](https://github.com/pjhampton/kibana-prometheus-exporter) 
+  is installed. You can run the following checks to verify
+  that {{app_name_short}} is exposing Prometheus-format metrics.
+
+   *  To determine if {{exporter_name}} is exposing metrics, run the following command:
```
[the remark this answers is not in the document]
Comment (jsirianni) [reply integrations/kibana/documentation.yaml]: > This block doesn't really make it clear that you have to manually install this exporter plugin.
> 
> Suggest rewording this a bit. (Plus, sentence at lines 11-12 is redundant with line 14. And the "command" after line 14 is actually two separate commands, so we should describe them separately. Something like this is what we've done in other variants:
> 
> ```
> additional_prereq_info: |
>   {{app_name_short}} exposes Prometheus-format metrics only after you install
>   {{exporter_name}} plugin.
> 
>   0. To install the plugin, follow the [installation instructions]({{exporter_repo_url}}).
>   
>   0. Set up port forwarding by running the following command:
> 
>       <pre class="devsite-click-to-copy">
>       kubectl -n {{namespace_name}} port-forward {{pod_name}} 5601
>       </pre>
>     
>   0. Verify that the exporter is emitting metrics at the expected endpoint by using your
>        browser or the \`curl\` utility to access the \`localhost:5601/_prometheus/metrics\` endpoint.
> ```
> 
> (However, looking at the installation instructions for this plugin, it's not clear to me that the port-forwarding step is necessary. Their instructions configure the port and metrics path in the config and tell you to look at http://localhost:5601/_prometheus/metrics, so maybe all you need to say here is something like this:
> 
> ```
> additional_prereq_info: |
>   {{app_name_short}} exposes Prometheus-format metrics only after you install
>   {{exporter_name}} plugin. To install the plugin and verify that metrics are being 
>   emitted, follow the [exporter installation instructions]({{exporter_repo_url}}).
> ```
> 
> Or we replicate their config addition for the exporter job as "config_mods" ?

Thanks, I have implemented this change. The port forwarding step is required for testing the metrics plugin. Without forwarding a port to your system, `localhost:5601` will not work.

> @mtabasko 's feedback on wording is great, but to take a step back --
> 
> I tried testing these instructions by setting up my own kibana instance on a toy cluster via [Elastic's quickstart instructions](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-deploy-kibana.html) and it's not immediately obvious even to me how to follow https://github.com/pjhampton/kibana-prometheus-exporter to install the exporter.
> 
> Can we give further hints?
> 
>     * should I be creating their own docker image, hosting that, and then create a deployment in the cluster on it, something like:
> 
> 
> ```
> apiVersion: apps/v1
> kind: Deployment
> metadata:
>   name: kibana-prometheus-exporter
> spec:
>   template:
>     spec:
>       containers:
>       - name: kibana
>          image: ...
> ```
> 
> although I see that [there is no official published image](https://www.github.com/pjhampton/kibana-prometheus-exporter/issues/263) which makes this harder than say, our [Elasticsearch installation instructions](https://cloud.google.com/stackdriver/docs/managed-prometheus/exporters/elasticsearch).
> 
>     * or should I be modifying their elastic-operator's Kibana CRD as per [this](https://www.github.com/pjhampton/kibana-prometheus-exporter/issues/137)?
>       See our [Zookeeper](https://cloud.google.com/stackdriver/docs/managed-prometheus/exporters/zookeeper) and [Consul](https://cloud.google.com/stackdriver/docs/managed-prometheus/exporters/consul) docs as examples of us documenting how to change config an existing application to emit metrics.
> 
> 
> Either way, to reach the same bar of user-friendliness as our other docs, let's elaborate more on how to install the exporter before jumping to "this is how you verify the metrics are being emitted at localhost:5601/_prometheus/metrics"

The exporter's readme does indicate that a container image needs to be built to include the plugin. I do agree that this presents some unfortunate overhead (building, pushing, maintaining your own image).

The example you shared is could work, but is incomplete. The init container (and kibana container) would need to share an `emptyDir` volume, which would allow the init container to persist the installed plugin. This would probably work, but can get tricky if the kibana image being used already contains plugins. Using a volume will cause anything in that directory (baked into the image) to become unavailable. Additionally, the plugin would be installed during every pod's startup, which could fail if there was an issue with the plugin download / network.

In my opinion, building an image is the most reliable solution. I am happy to try the init container path if you think it is more appropriate.

[the remark this answers is not in the document]
Comment (yqlu) [reply integrations/kibana/documentation.yaml]: Ok, thanks, SGTM! Let's take a stab at writing instructions to build an image. 

We don't have any existing GMP exporter docs that feature this, but I do see that [the Varnish PR](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/pull/431/files#diff-b27b4afe5022e3dda7b6e555ca974ecdb2880eaf537bc5a543eae7b59e451b58R56) (which we haven't gotten to reviewing yet) also has instructions for building your own container, so let's be consistent across both of them.

File: integrations/kibana/prometheus_metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+platforms:
+  - type: GKE
+    launch_stage: GA
+    detections:
+      - characteristic_metric:
+          metric_type: prometheus.googleapis.com/kibana_millis_uptime/gauge
+    exporter_metadata:
+      name: Kibana Prometheus Exporter
+      doc_url: https://github.com/pjhampton/kibana-prometheus-exporter
+      minimum_supported_version: "8.0.0"
```
[the remark this answers is not in the document]
Comment (yqlu) [reply integrations/kibana/prometheus_metadata.yaml:10]: Sorry, follow-up question after running through the instructions as a user. Just double-checking if this is true? I was able to run this (and populate the dashboard) with 7.17.3 kibana against 7.17.3 elasticsearch..

**Review by yqlu — COMMENTED** [review]
Appreciate the docker image instructions. I was able to follow the instructions as a user and get metrics ingested / my Kibana dashboard populated with numbers. Just a couple questions left.

File: integrations/kibana/prometheus_metadata.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (algchoo) [reply integrations/kibana/prometheus_metadata.yaml:10]: This is incorrect. Looking further, the exporter documentation explains that [the version of the exporter must match the version of kibana.](https://github.com/pjhampton/kibana-prometheus-exporter#releases) Should we exclude the `minimum_supported_version` or may we just replace the version number with a description?

```
"8.0.0" => "Version must match Kibana" 
```

File: integrations/kibana/documentation.yaml [hunk 3]
```diff
@@ -22,12 +22,26 @@ additional_prereq_info: |
...
   ```yaml
+  apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: kibana
+  data:
+    kibana.yml: |
+      server.name: kibana
+      server.host: "0.0.0.0"
+      elasticsearch.hosts: http://es-kibana:9200
```
Comment (yqlu) [inline integrations/kibana/documentation.yaml:33]: Let's make this more explicitly a placeholder

Like
```
  # Update this with credentials to match your Elasticsearch instance
  elasticsearch.hosts: http://username:password@elasticsearch-service-name:9200
```

**Review by yqlu — APPROVED** [review]
